core/src/main/java/org/apache/carbondata/core/datastore/page/SafeFixLengthColumnPage.java

```diff
@@ -165,9 +165,28 @@ public double getDouble(int rowId) {
     return doubleData[rowId];
   }
 
-  @Override
-  public BigDecimal getDecimal(int rowId) {
-    throw new UnsupportedOperationException("invalid data type: " + dataType);
+  @Override public BigDecimal getDecimal(int rowId) {
```

Comment:
I think it is better to add a DecimalColumnPage instead of using FixLengthColumnPage

Reply:
Please correct me If I am wrong....From my opinion it will not be any advantage to add DecimalColumnPage as we are not doing anything extra if we introduce DecimalColumnPage.
Basically we are storing values either as fixed or variable length and decimal values also falls in these 2 categories. So according to current design I believe it may not be required to introduce 1 more type of page until and unless we have pages for all other datatypes separately.
Please share your suggestions.

Reply:
I think adding converter inside the column page make it a bit complex. I think there are two mays to make the code clean:
1. Store BigDecimal array in the ColumnPage and do the conversion to int or long when come to encoding part, by implementing a ColumnPageValueConverter
2. Create DecimalColumnPage class and add the conversion logic in DecimalColumnPage. This logic is the extra logic for decimal only. If you do like this, we can consider to rename `FixLengthColumnPage` to `PrimitiveColumnPage` and `VarLengthColumnPage` to `StringColumnPage` and the 3rd one is `DecimalColumnPage`. The responsibility of each class will be more clear.